src/server_manager/model/survey.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export interface Surveys {
+  // Returns a Survey for `surveyId`, configured to be displayed after `promptImpressionDelayMs`,
+  // and only before a date `displayBefore`.
+  requestSurvey(surveyId: string, promptImpressionDelayMs: number, displayBefore?: Date): Survey;
+}
+
+export interface Survey {
```

Comment:
We'll be able to delete this interface with the suggestions below.

Reply:
Done.